synapse/lib/nexus.py

```diff
@@ -492,9 +534,8 @@ async def runMirrorLoop(self, proxy):
                 async for item in genr:
 
                     if self.readonly:
-                        logger.error('Unable to consume Nexus events when readonly is set')
-                        await self.client.fini()
-                        return
+                        await self.waitfini(timeout=2)
+                        break
```

Comment:
Is it worth adjusting this to wait on `self.ready` or `fini` rather than polling with `waitfini`?

Reply:
I think `self.ready` has a bit of a conflated use case ( we've still got logic that sets it only once we enter the real-time window and notes about bumping telepath clients until then ) so i avoided that one for now. I figured a fini-safe 2 second poll should be fine to account for readonly status changing etc. The proxy will be fini'd on a reconnect which would break out of the loop anyway. Open to idea, but was trying to avoid refactoring the use of `self.ready` in this PR :)